lib/stat_tracker.rb

```diff
@@ -56,12 +53,102 @@ def percentage_visitor_wins
     (away_win?.to_f / @games.size).round(2)
   end
 
+  def total_goals_in_game
+    @games.map do|game| 
+      game.away_goals + game.home_goals
+    end
+  end
+
+  def count_of_games_by_season
+    games_by_season = {}
+    @games.each do |game|
+      if games_by_season.include?(game.season_id.to_sym)
+        games_by_season[game.season_id.to_sym] += 1 
+      else
+        games_by_season[game.season_id.to_sym] = 1 
+      end
+    games_by_season
+  end
+
+  def count_of_goals_by_season
+    goals_by_season = {}
+    @games.each do |game|
+      if goals_by_season.include?(game.season_id.to_sym)
+        goals_by_season[game.season_id.to_sym] += ([game.away_goals.to_f]+[game.away_goals.to_f])
+      elsif !goals_by_season.include?(game.season_id.to_sym)
+        goals_by_season[game.season_id.to_sym] = ([game.away_goals.to_f]+[game.home_goals.to_f])
+      end 
+    goals_by_season
+  end
+
+  def average_goals_per_game
+    (total_goals_in_game.sum / @games.count.to_f).round(2)
+  end
+
+  def average_goals_by_season
+    average_goals = {}  
+    count_of_goals_by_season.each do|season, goals|
+      count_of_games_by_season.each do|season_id, games|
+        if season_id == season
+        average_goals[season_id] = (goals/games).round(2) 
+    average_goals
+  end
+
+end
+
+
   # LEAGUE STATS
 
   def count_of_teams
     @teams.count
   end
 
+
+  def team_goals(home_or_away)
+    team_goals = {}
+    @game_teams.each do |game_team|
+      if team_goals[game_team.team_id.to_sym] != nil && game_team.home_or_away == home_or_away
+        team_goals[game_team.team_id.to_sym].push(game_team.goals.to_i)
+      elsif  game_team.home_or_away == home_or_away
+        team_goals[game_team.team_id.to_sym] = [game_team.goals.to_i]
+      end
+    end
+    team_goals
+  end
+
+  def average_goals(team_goals)
+    team_average = {}
+    team_goals.each do |team_id, goals_per_game|
+      team_average[team_id] = (goals_per_game.sum.to_f / goals_per_game.size.to_f).round(3)
+    end
+    team_average
+  end
+
+  def team_goal_average(home_or_away)
+      team_goals = team_goals(home_or_away)
+      average_goals(team_goals)
+    end
+  end
+
+  def find_team_name(team_element)
+    @teams.each do |team|
+      if team_element[0] == team.team_id
+        team_name = team.team_name
+      end
+    end
+  end
+
+  def highest_scoring_home_team
```

Comment:
Hey, I doubled checked the method assignment sheet and you were supposed to do the visitor methods, I did the ones for the home team and have already pushed those up, It should be an easy fix just changing it to away instead of home.